app/models/calendars/calendar_type.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class CalendarType < ApplicationRecord
+  # relations
+  has_many :calendar_title_types, foreign_key: :type_id, dependent: :destroy
+  has_many :titles, through: :calendar_title_types
+  accepts_nested_attributes_for :calendar_title_types, reject_if: :all_blank, allow_destroy: true
```

Comment:
AcademicCalendar modelinde `belongs_to :calendar_type` var ama burda karşılığı yok. Anlamsız diye eklemediysen sorun yok, ama gözden kaçtıysa diye not düşüyorum.